quintilesims/eks-sso
# Frontend for eks-sso service

Adds front-end ui portion of the eks-sso service.

**How should this be tested:**
- refer to develop.md to run the service locally
- test functionality working locally:
  1. login with ldap credentials
  2. add IAM credentials for account with EKS cluster
  3. copy generated KubeConfig locally and confirm you can connect to the EKS cluster
  4. confirm a namespace has been created (namespace name should match the IAM user name)
- check functionality locally and by deploying to a EKS cluster

This is my first iteration at building a production service in react. So advice about best patterns and practices that aren't being followed would also be helpful.

Any issues that aren't critical to the functionality yet require significant amount of refactoring would be better spun off as a new issue. For example, writing react unit tests will be spun off into a separate issue.

Closes: #10
